packages/block-library/src/navigation/view.js

```diff
@@ -1,74 +1,34 @@
-// Open on click functionality.
-function closeSubmenus( element ) {
-	element
-		.querySelectorAll( '[aria-expanded="true"]' )
-		.forEach( function ( toggle ) {
-			toggle.setAttribute( 'aria-expanded', 'false' );
-		} );
-}
+/**
+ * Internal dependencies
+ */
+import wpx from './wpx';
 
-function toggleSubmenuOnClick( event ) {
-	const buttonToggle = event.target.closest( '[aria-expanded]' );
-	const isSubmenuOpen = buttonToggle.getAttribute( 'aria-expanded' );
-
-	if ( isSubmenuOpen === 'true' ) {
-		closeSubmenus( buttonToggle.closest( '.wp-block-navigation-item' ) );
-	} else {
-		// Close all sibling submenus.
-		const parentElement = buttonToggle.closest(
-			'.wp-block-navigation-item'
-		);
-		const navigationParent = buttonToggle.closest(
-			'.wp-block-navigation__submenu-container, .wp-block-navigation__container, .wp-block-page-list'
-		);
-		navigationParent
-			.querySelectorAll( '.wp-block-navigation-item' )
-			.forEach( function ( child ) {
-				if ( child !== parentElement ) {
-					closeSubmenus( child );
+wpx( {
+	core: {
+		navigation: {
+			openMenu: ( { context: { navigation } } ) => {
+				navigation.open = true;
+				navigation.previousElementWithFocus =
+					window.document.activeElement;
+			},
+			closeMenu: ( { context: { navigation } } ) => {
+				navigation.open = false;
+			},
+			isMenuOpen: ( { context: { navigation } } ) => navigation.open,
+			isMenuClosed: ( { context: { navigation } } ) => ! navigation.open,
+			addFirstElementToContext: ( { context: { navigation }, ref } ) => {
+				navigation.firstMenuElement = ref;
+			},
+			focusFirstElement: async ( { context: { navigation }, tick } ) => {
+				if ( navigation.open ) {
+					await tick(); // We need to wait until the DOM is updated.
```

Comment:
This won't be needed once this is merged: https://github.com/preactjs/signals/pull/290